gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/handler/Rexster.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.tinkerpop.gremlin.server.handler;
+
+import io.netty.channel.Channel;
+import org.apache.tinkerpop.gremlin.driver.message.ResponseMessage;
+import org.apache.tinkerpop.gremlin.server.Context;
+
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledFuture;
+
+/**
+ * The {@code Rexster} interface is essentially a form of "worker", named for Gremlin's trusty canine robot friend
```

Comment:
Is it a form of "worker" or form of "work"? Since it implements Runnable, it seems like Rexter is a unit of work that is processed by RexterExecutorService.

This implements runnable (which signifies that it is a standalone task) and yet it's implementations have a addTask() method and contains executorService as it's member?

Here's a suggestion, let me know what you think about it:
- GremlinOpExecutor is a service backed which executes a GremlinOpTask. 
- GremlinOp can either be TransactionOp or QueryOp etc.
- GremlinOpExecutor has lifecycle states which begins at server start up and ends on server shutdown.
- GremlinOpTask to GremlinOpExecutor are provided from a Netty Event loop thread pool (which we shouldn't block) and hence, any underlying implementation should delegate to it's own thread pool ASAP.
- Interface SessionExecutor containing methods like setSessionThread etc.
- The concrete implementation, MultiTaskGremlinOpExecutor can execute multiple GremlinOps at the same time (using a queue and worker thread pool underneath) and it also implements SessionExecutor. 
- GremlinOpExecutor should have methods such as:
   - Future executeGremlinOpAsync
   
   
I am suggesting the above model so that we can get some unification in vocabulary used in the code base. Currently we have terms such as task, job, rexter, worker, executor, graph operation etc. sued across the code base without any concrete meaning to them which makes it difficult to read code.

Reply:
ok - I'll not indulge myself this time with fun naming schemes 😄